ndcube/tests/test_ndcube.py

```diff
@@ -520,3 +520,49 @@ def test_initialize_from_ndcube(ndcube_3d_l_ln_lt_ectime):
     assert ec.mapping == ec3.mapping
     assert np.allclose(ec.wcs.pixel_to_world_values(1), ec3.wcs.pixel_to_world_values(1))
     assert ec is not ec3
+
+
+def test_resampling(ndcube_4d_ln_l_t_lt, wcs_4d_lt_t_l_ln, wcs_3d_lt_ln_l):
+    target_wcs_header = wcs_4d_lt_t_l_ln.low_level_wcs.to_header()
+    target_wcs_header['CDELT3'] = 0.1   # original value = 0.2
+    target_wcs = astropy.wcs.WCS(header=target_wcs_header)
+    shape_out = (5, 20, 12, 8)
+
+    resampled_cube = ndcube_4d_ln_l_t_lt.reproject(target_wcs, shape_out)
+
+    assert ndcube_4d_ln_l_t_lt.data.shape == (5, 10, 12, 8)
+    assert resampled_cube.data.shape == (5, 20, 12, 8)
+
+    # Raises an exception if given target_wcs is not compatible with the NDCube
+    with pytest.raises(Exception):
+        _ = ndcube_4d_ln_l_t_lt.reproject(wcs_3d_lt_ln_l, shape_out)
```

Comment:
Could you move these lines to another test that just tests that the correct exception is raised when the WCS is incompatible?